### PR TITLE
Feature/cleanup code

### PR DESCRIPTION
**Pull request by jensarps**
This is a replacement PR for artoolkit/jsartoolkit5#16.

I'm terribly sorry.

### CODE REVIEW

**Comment by jeromeetienne** [comment]
+1 globally, except the ```var``` being centrallized

**Comment by jensarps** [comment]
The commit about the var declaration issues (38a2ea19a0f8398cd28ba53c290ba11f323df88a) was about removing duplicate var declarations. The alternative would be to keep the first declarations and remove the following, e.g.:

```javascript
// keep this
var visible = this.trackPatternMarkerId(-1);

// and afterwards, remove the duplicate var statement:
visible = false;
```

Personally, I find this approach to be confusing, so I prefer to declare the vars separately. Or, use different var names.

**Comment by ThorstenBux** [comment]
Hi @jensarps I know it has been a long while since you opened that PR. However, as I currently clean-up the PR I'd like to get yours it into master. If you read that and are still interested maybe you can update your branch and we can merge this. Thanks

**Comment by jensarps** [comment]
Hey, good to see someone's still working on this :) Sure, I'll have a look at it this weekend.

**Comment by jensarps** [comment]
Sorry for the delay, ready to go now :)

What happened to #15 and #19? 

API docs could be easily created now. And a meaningful package.json and versioned releases seem quite mandatory to me – also given that AR.js builds on top of this and did gain quite some traction the other day. You could make use of npm to distribute jsartoolkit, allowing consumers like AR.js to pull it in via a package manager instead of manually copying it into the project :) /cc @jeromeetienne

**Comment by ThorstenBux** [comment]
Hi @jensarps thanks a lot for doing that 👍 . I don't know what happened to #15 and #19 but it looks like you closed #19 yourself :).

The next weeks I'll look into packaging a npmjs package for jsartoolkit and writing a package.json with verisons, etc. I'm more then happy if you would like to contribute on that, as I haven't created a npmjs package yet.

Thanks again

File: js/artoolkit.api.js [hunk 1]
```diff
@@ -22,7 +22,6 @@
 		@param {ARCameraParam | string} camera The ARCameraParam to use for image processing. If this is a string, the ARController treats it as an URL and tries to load it as a ARCameraParam definition file, calling ARController#onload on success. 
 	*/
 	var ARController = function(width, height, camera) {
-		var id;
```
[the remark this answers is not in the document]
Comment (jensarps) [reply js/artoolkit.api.js]: It's unused.

**Comment by jensarps** [comment]
Sure, you're welcome :)

I'll gladly help out regarding the NPM package – but this will be more about long-term actions, decisions and quality assurance than about the actual publishing. This includes writing tests, enforcing a coding style, deciding on a branching model, keeping dependencies up-to-date, having a small but defined build process or making sure to always follow semver. But, yeah, I can help with that :)

Regarding #19, I could update the `gh-pages` branch with the docs and open another PR for that.

**Pull request merged**